static/templyte/src/app/download-template/download-template.component.ts

```diff
@@ -90,9 +95,28 @@ export class DownloadTemplateComponent implements OnInit {
                     'cancelable': false
                 });
                 linkElement.dispatchEvent(clickEvent);
+                component.templateDirectoryInfoDatabaseRef.child('/templateNumDownload').transaction(function(snapshot) {
+                    return snapshot+1;
+                });
+                component.templateDirectoryInfoRef.update({
+                    'templateLastDownloadDate': formatDate(new Date(), 'yyyy/MM/dd', 'en')});
```

Comment:
Why are we storing dates as strings? - We're not going to be able to index on these correctly.